### DISCUSSION TITLE
How to clear selection items after selection operations?

### DISCUSSION THREAD

**Discussion by frixaco** [sayanarijit/xplr#675] [Q&A] [answered]
Thank you very much for this amazing tool! I finally made full switch to `xplr` and it has been a fun journey except for one little thing.

There has been a few times, when I forget to clear selection list before doing any other operations and accidentally end up deleting selected files. Although, this is a skill issue which requires getting used to, I'd prefer `xplr` auto clear selection list, because I never encountered a case when I'd need selection items after e.g. copying, moving them over to a place once.

I took a look at what `Ctrl+u` does:
```
["u"] = {
        help = "clear selection",
        messages = {
          "ClearSelection",
          "PopMode",
        },
      },
```
and was thinking to "append" this action after each selection operation (Ctrl+c, Ctrl+m, etc.) somehow.

Is there any other "better" ways to achieve that?

**Comment by sayanarijit** [comment, marked as answer]
There is some safety in place. Like when you delete, if you're using trash-cli.xplr, you can easily recover by typing dr. If you're not, xplr will prompt you displaying the list of paths. For which, I think keeping the list isn't that problematic. Upside is that you get to do multiple operations such as
- delete after copy (safer move alternative for inter filesystem operations)
- perform other actions on paths that weren't moved after move operations
- custom actions that perform multiple operations on the same files (by keeping the behaviour consistent with default)

It is also compatible with map.xplr's behaviour which has clear upsides of keeping the list.

If don't prefer this, you can clear the list by adding "ClearSelection" as the last message, for any action you want. This is the cleanest way.

**Comment by frixaco** [comment]
Here's the config to enable auto selection clear (only for copy and move):
```
xplr.config.modes.builtin.selection_ops.key_bindings.on_key["c"] = {
	help = "copy here",
	messages = {
		xplr.config.modes.builtin.selection_ops.key_bindings.on_key["c"].messages[1],
		"ClearSelection",
	},
}

xplr.config.modes.builtin.selection_ops.key_bindings.on_key["m"] = {
	help = "move here",
	messages = {
		xplr.config.modes.builtin.selection_ops.key_bindings.on_key["m"].messages[1],
		"ClearSelection",
		"PopMode",
	},
}
```
